artifacts/reader.py

```diff
@@ -72,6 +72,10 @@ def _ReadArtifactDefinition(self, yaml_definition):
     if not yaml_definition:
       raise errors.FormatError(u'Missing YAML definition.')
 
+    different_keys = set(yaml_definition) - definitions.TOP_LEVEL_KEYS
+    if different_keys:
+      raise errors.FormatError(u'Bad keys: {0!s}'.format(different_keys))
```

Comment:
So the format spec is currently not strict about having additional keys. I opt to have GRR be more tolerant in handling these but I'm fine with having a "strict" format check. However we should not point out these "keys" as "bad", since they are merely undocumented.